src/App.js

```diff
@@ -34,50 +34,130 @@ class App extends Component {
+        this.tuitionFilter = this.tuitionFilter.bind(this);
+        this.satFilter = this.satFilter.bind(this);
+        this.actFilter = this.actFilter.bind(this);
+        this.rankingFilter = this.rankingFilter.bind(this);
+
     }
     componentWillMount() {
         var childState = Object.assign({}, this.state);
-       // childState.data = data;
         this.child = React.cloneElement(this.props.children, childState);
     }
     componentWillReceiveProps(props){
         var childState = Object.assign({}, this.state);
-        //childState.data = data;
         this.child = React.cloneElement(this.props.children, childState);
-        //this.forceUpdate();
     }
     setZip(event){
 
         this.handleChange(event);
     }
     changeValue(e){
         this.state.filter.tuition = e.target.value;
-        console.log(this.state);
     }
     applyFilter(e){
         e.preventDefault();
+        if(/(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.state.filter.zip)){
+            var url = "https://maps.googleapis.com/maps/api/geocode/json?address="+ encodeURIComponent(this.state.filter.zip);
+            fetch(url) //download the data
+            .then(function(res) { return res.json(); })
+            .then((datas) =>{
+                return this.setZipLocation(datas.results);
+            })
+            .then(this.satFilter)
+            .then(this.actFilter)
+            .then(this.rankingFilter)
+            .then(this.tuitionFilter);
+        }else{
+            this.tuitionFilter(this.rankingFilter(this.actFilter(this.satFilter(data))));
+        }
+    
+    }
+    satFilter(datas){
 
-        var url = "https://maps.googleapis.com/maps/api/geocode/json?address="+ encodeURIComponent(this.state.filter.zip);
-        fetch(url) //download the data
-      .then(function(res) { return res.json(); })
-      .then((datas) =>{
-        this.setZipLocation(datas.results);
-      });
-      var newData = data.filter((obj)=>{
-            return obj['2014.cost.tuition.out_of_state'] >= this.state.filter.tuition[0] && obj['2014.cost.tuition.out_of_state'] <= this.state.filter.tuition[1];
-        });
-        this.setState({"data": newData});
-      var childState = Object.assign({}, this.state);
-      //childState.data = data;
-      this.child = React.cloneElement(this.props.children, childState);
-        this.forceUpdate();
+        if(this.state.filter.SAT !== ""){
+            return datas.filter((obj)=>{
+                return this.state.filter.SAT>=obj["2014.admissions.sat_scores.average.overall"];
+            });
+        }else{
+            return datas;
+        }
+    }
+    actFilter(datas){
+
+        if(this.state.filter.ACT !== ""){
+            return datas.filter((obj)=>{
+                return this.state.filter.ACT>=obj["2014.admissions.act_scores.midpoint.cumulative"];
+            });
+        }else{
+            return datas;
+        }
+        
+    }
+    rankingFilter(datas){
+
```

Comment:
Don't put blank lines at the beginning and end of functions. Do put blank lines before functions.